jetbrains-core/src/software/aws/toolkits/jetbrains/services/telemetry/TelemetryService.kt

```diff
@@ -24,13 +24,25 @@ import java.time.Instant
 import java.util.concurrent.CompletableFuture
 import java.util.concurrent.atomic.AtomicBoolean
 
-interface TelemetryService : Disposable {
+abstract class TelemetryService(private val publisher: TelemetryPublisher, private val batcher: TelemetryBatcher) : Disposable {
     data class MetricEventMetadata(
         val awsAccount: String = METADATA_NA,
         val awsRegion: String = METADATA_NA
     )
 
-    fun record(metricEventMetadata: MetricEventMetadata, buildEvent: MetricEvent.Builder.() -> Unit = {}): MetricEvent
+    private val isDisposing = AtomicBoolean(false)
+    private val startTime: Instant
+
+    init {
+        // TODO this startup stuff should be moved to a global startup task instead of in the constructor FIX_WHEN_MIN_IS_193
+        // The auto generated telemetry cannot be used here. It tries to get the instance while
+        // constructing it which causes a circular dependency issue.
+        record("session_start").also {
```

Comment:
We can remove session_start

Reply:
I asked yesterday, but did not get a clear answer...will ask again since I would like to clean that up and remove even more code